.pipelines/templates/.builder-release-template.yaml

```diff
@@ -111,58 +111,114 @@ steps:
+          -e CONTAINER_RUNTIME=${CONTAINER_RUNTIME:-"docker"} \
+          -e OS_VERSION=${OS_VERSION} \
+          -e OS_SKU=${OS_SKU} \
+          -e OS_TYPE=\'Linux\' \
+          -e IMG_SKU=${IMG_SKU} \
+          -e MODE=$(MODE) \
+          -e VHD_DEBUG=${VHD_DEBUG} \
+          -e FEATURE_FLAGS=$(FEATURE_FLAGS) \
+          -e SIG_GALLERY_NAME=${SIG_GALLERY_NAME} \
+          -e SIG_IMAGE_NAME=${SIG_IMAGE_NAME} \
+          -e SIG_IMAGE_VERSION=${SIG_IMAGE_VERSION} \
+          -e ENABLE_FIPS=${ENABLE_FIPS} \
+          -e ARCHITECTURE=${ARCHITECTURE} \
+          -e ENABLE_TRUSTED_LAUNCH=${ENABLE_TRUSTED_LAUNCH} \
+          -e SGX_INSTALL=${SGX_INSTALL} \
+          -e ENABLE_CGROUPV2=${ENABLE_CGROUPV2} \
+          -e GIT_BRANCH=$(Build.SourceBranch) \
+          ${CONTAINER_IMAGE} make -f packer.mk test-building-vhd" \
+          "docker run --rm \
+          -v ${PWD}:/go/src/github.com/Azure/AgentBaker \
+          -w /go/src/github.com/Azure/AgentBaker \
+          -e SUBSCRIPTION_ID=$(SUBSCRIPTION_ID) \
+          -e PKR_RG_NAME=${PKR_RG_NAME} \
+          -e MODE=${MODE} \
+          -e DRY_RUN=${DRY_RUN} \
+          -e OS_TYPE=\'Linux\' \
+          -e AZURE_RESOURCE_GROUP_NAME=${AZURE_RESOURCE_GROUP_NAME} \
+          -e IMAGE_NAME=${IMAGE_NAME} \
+          -e CAPTURED_SIG_VERSION=${captured_sig_version} \
+          -e IMPORTED_IMAGE_NAME=${IMPORTED_IMAGE_NAME} \
+          -e SIG_GALLERY_NAME=${SIG_GALLERY_NAME} \
+          -e SIG_IMAGE_NAME=${SIG_IMAGE_NAME} \
+          -e ARCHITECTURE=${ARCHITECTURE} \
+          ${CONTAINER_IMAGE} make -f packer.mk cleanup"
+
+        gnu_exit_code=$?
+
+        for dir in results/*; do
+          job_number=$(basename "$dir")
+          new_name=$(awk -v job="$job_number" '$1 == job {print $11}' log.txt)
+          mv "$dir" "results/$new_name"
+        done
+
+        for target in scanning-vhd test-building-vhd cleanup; do
+          echo "##[group]${target}"
+          echo "##[group]Output"
+          echo "##[group]stdout"
+          cat results/${target}/stdout
+          echo "##[endgroup]"
+          echo "##[group]stderr"
+          cat results/${target}/stderr
+          echo "##[endgroup]"
+          echo "##[endgroup]"
+          echo "##[group]Results"
+          exit_status=$(awk -F'\t' -v target="$target" '$9 ~ target {print $7}' log.txt)
+          if [ $exit_status -eq 0 ]; then
+            echo "${target} succeeded with exit code $exit_status"
+          else
+            echo "${target} failed with exit code $exit_status"
+          fi
+          echo "##[endgroup]"
+          echo "##[endgroup]"
+        done
+
+        if [ ${gnu_exit_code} -ne 0 ]; then
+          echo "The following targets failed:"
+          awk -F'\t' '$7 != 0 {print $9}' log.txt
+          exit 1
+        else
+          echo "All make targets succeeded"
+        fi
+    condition: always()
+    displayName: Test, Scan, and Cleanup
```

Comment:
maybe something else you can think about, and maybe we already decided to head in this direction but I'm not sure off the top of my head - could we combine the testing/scanning steps such that they use the same underlying VM?

Reply:
I considered that but was concerned about the resource usage on the vm between several az run commands, the scanning executable, test script, and the cleanup. 

Though I suppose we could run cleanup on the agent, and test and scan on the same vm. Would be willing to try but not sure if thats what you are doing now.

Reply:
`Though I suppose we could run cleanup on the agent, and test and scan on the same vm. Would be willing to try but not sure if thats what you are doing now.`

I meant this

basically combine the test/scan steps into a single step, executed on the same underlying VM in parallel (not agent VM)

then run that ^ step in parallel with cleanup

don't worry about it though, that's probably something we can follow-up with in another PR